Sources/TuistDoc/SwiftDocServer.swift

```diff
@@ -0,0 +1,130 @@
+import Dispatch
+import Foundation
+import Signals
+import Swifter
+import TSCBasic
+import TuistSupport
+
+public protocol SwiftDocServing {
+    /// Base url for the server
+    var baseURL: String { get }
+
+    /// Serves the documentation at a given path
+    /// - Parameters:
+    ///   - path: Path to the folder containing the documentation
+    ///   - port: Port to use for hosting the website
+    func serve(path: AbsolutePath, port: UInt16) throws
+}
+
+public final class SwiftDocServer: SwiftDocServing {
+    private static let index = "index.html"
+    private static var temporaryDirectory: AbsolutePath?
+
+    /// Utility to manipulate files
+    private let fileHandling: FileHandling
+
+    /// Utility to open files
+    private let opener: Opening
+
+    /// HTTPServer from Swifter
+    private var server: HttpServer?
+
+    /// Semaphore to block the execution
+    private var semaphore: DispatchSemaphore?
+
+    public let baseURL: String = "http://localhost"
+
+    public init(fileHandling: FileHandling = FileHandler(),
+                opener: Opening = Opener())
+    {
+        self.fileHandling = fileHandling
+        self.opener = opener
+    }
+
+    public func serve(path: AbsolutePath, port: UInt16) throws {
+        SwiftDocServer.temporaryDirectory = path
```

Comment:
If we consider `path` is a temporary directory, the implementation of this method has knowledge about the directory that is being passed, which is not a good practice. What you can do is extracting the logic using semaphores and deleting the directory into the class that generates the documentation and starts the server. That's the one that has the knowledge on the temporary character of the server and therefore should be the one making decisions about what to do if there's a signal received.